src/dvc_data/hashfile/checkout.py

```diff
@@ -108,6 +107,11 @@ def _checkout_file(
                 prompt=prompt,
             )
     else:
+        try:
+            fs.remove(path)
+        except FileNotFoundError:
+            pass
+
         link(cache, cache_path, fs, path)
```

Comment:
This is the only place where we don't call `_remove()`. But since it happens when `change.old.oid` is None, that means the file does not exist locally before hand.

But I replaced with equivalent just to be safe. (As I am not able to trigger this piece of code).

Reply:
I have dropped `fs.remove()` from here. It does not seem like we need here.